src/cli/src/consume/mod.rs

```diff
@@ -51,9 +48,17 @@ pub struct ConsumeOpt {
     #[structopt(short, long)]
     pub key_value: bool,
 
-    /// Offsets can be positive or negative. (Syntax for negative offset: --offset="-1")
-    #[structopt(short, long, value_name = "integer")]
-    pub offset: Option<i64>,
+    /// Consume records starting X from the beginning of the log (default: 0)
+    #[structopt(short = "B", value_name = "integer", conflicts_with_all = &["offset", "tail"])]
+    pub from_beginning: Option<Option<u32>>,
```

Comment:
why is this option of option?

Reply:
There are three states:

- `fluvio consume <topic>` (no `--from-beginning`): consume from the end (corresponds to `None`)
- `fluvio consume <topic> --from-beginning`: consume from the very beginning (corresponds to `Some(None)`)
- `fluvio consume <topic> --from-beginning=10`: consume starting 10 after the beginning (`Some(Some(offset))`)

Reply:
Same pattern goes for `--tail`